src/engine/enginebuffer.cpp

```diff
@@ -705,6 +713,13 @@ void EngineBuffer::slotControlSlip(double v)
     m_slipEnabled = static_cast<int>(v > 0.0);
 }
 
+void EngineBuffer::slotControlSlipCancel(double v)
+{
+    if (load_atomic(m_slipEnabled)) {
```

Comment:
This might be cause a race condition. 
I think it will help to remove such conditions if we just collect the Cancel event here. 
And move all other logic to the engine thread. 

Something like: 
```
void EngineBuffer::slotControlSlipCancel(double v)
{
    if (v > 0.0) {
         m_slipCancelled = 1;
    }
}
```